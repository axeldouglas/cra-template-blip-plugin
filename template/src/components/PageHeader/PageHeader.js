import React from 'react';
import Proptypes from 'prop-types';

const COLOR_NEUTRAL_DARK_ROOFTOP = '#505F79';
const COLOR_NEUTRAL_MEDIUM_CLOUD = '#8CA0B3';

const PageHeader = ({
    title,
    icon,
    is_informative = false,
    help_text,
    is_back_navigation = false,
    onBackPressed,
    relatedOptions
}) => {
    const renderInfoTitle = () => (
        <div className="pointer" data-testid="page-header-tooltip">
            <bds-tooltip tooltip-text={help_text} position="right-center">
                <bds-icon
                    name="info"
                    theme="solid"
                    color={COLOR_NEUTRAL_MEDIUM_CLOUD}
                />
            </bds-tooltip>
        </div>
    );

    return (
        <div className="flex flex-row items-center-ns justify-between w-100 pv3 mt2 bb bp-bc-neutral-medium-wave">
            <div className="flex items-center">
                {is_back_navigation && (
                    <div
                        className="pointer mr1"
                        onClick={onBackPressed}
                        data-testid="page-header-back-icon"
                    >
                        <bds-icon
                            name="arrow-left"
                            color={COLOR_NEUTRAL_DARK_ROOFTOP}
                        />
                    </div>
                )}

                {!!icon && (
                    <div data-testid="page-header-icon">
                        <bds-icon
                            name={icon}
                            color={COLOR_NEUTRAL_DARK_ROOFTOP}
                        />
                    </div>
                )}

                <h2 className="f3 ml2 mr1 bp-c-neutral-dark-city">{title}</h2>

                {is_informative && !!help_text && renderInfoTitle()}
            </div>
            {!!relatedOptions && (
                <div className="flex items-center justify-end">
                    {relatedOptions}
                </div>
            )}
        </div>
    );
};

PageHeader.propTypes = {
    title: Proptypes.string,
    icon: Proptypes.string,
    is_informative: Proptypes.bool,
    help_text: Proptypes.string,
    is_back_navigation: Proptypes.bool,
    onBackPressed: Proptypes.func,
    relatedOptions: Proptypes.any
};

export default PageHeader;
