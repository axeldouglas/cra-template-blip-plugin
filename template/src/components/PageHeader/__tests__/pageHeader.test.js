import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';

import PageHeader from '../PageHeader';

describe('Page header component', () => {
    const handleClick = jest.fn();

    const mockRenderOptions = () => <>Hello Element</>;

    it('should render correctly', () => {
        render(
            <PageHeader
                title="Plugins"
                icon="plugin"
                is_informative={true}
                help_text="Tooltip Text Info"
                relatedOptions={mockRenderOptions()}
                is_back_navigation={false}
            />
        );
        expect(
            screen.getByRole('heading', {
                name: /plugins/i
            })
        ).toBeInTheDocument();

        expect(screen.queryByTestId('page-header-icon')).toBeInTheDocument();
        expect(screen.queryByTestId('page-header-tooltip')).toBeInTheDocument();

        expect(screen.getByText(/hello element/i)).toBeInTheDocument();

        expect(
            screen.queryByTestId('page-header-back-icon')
        ).not.toBeInTheDocument();
    });

    it('should handle click event', () => {
        render(
            <PageHeader
                title="Plugins"
                relatedOptions={mockRenderOptions()}
                is_back_navigation={true}
                onBackPressed={handleClick}
            />
        );
        const back_button_node = screen.queryByTestId('page-header-back-icon');
        fireEvent.click(back_button_node);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('should matches with snapshot', () => {
        const { container } = render(
            <PageHeader
                title="Plugins"
                icon="plugin"
                is_informative={true}
                help_text="Tooltip Text Info"
                relatedOptions={mockRenderOptions()}
                is_back_navigation={false}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
