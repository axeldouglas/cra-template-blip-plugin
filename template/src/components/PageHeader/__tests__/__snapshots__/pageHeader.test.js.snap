// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page header component should matches with snapshot 1`] = `
<div
  class="flex flex-row items-center-ns justify-between w-100 pv3 mt2 bb bp-bc-neutral-medium-wave"
>
  <div
    class="flex items-center"
  >
    <div
      data-testid="page-header-icon"
    >
      <bds-icon
        color="#505F79"
        name="plugin"
      />
    </div>
    <h2
      class="f3 ml2 mr1 bp-c-neutral-dark-city"
    >
      Plugins
    </h2>
    <div
      class="pointer"
      data-testid="page-header-tooltip"
    >
      <bds-tooltip
        position="right-center"
        tooltip-text="Tooltip Text Info"
      >
        <bds-icon
          color="#8CA0B3"
          name="info"
          theme="solid"
        />
      </bds-tooltip>
    </div>
  </div>
  <div
    class="flex items-center justify-end"
  >
    Hello Element
  </div>
</div>
`;
